src/test/regress/sql/undistribute_table.sql

```diff
@@ -117,6 +117,14 @@ SELECT undistribute_table('view_table');
 SELECT schemaname, viewname, viewowner, definition FROM pg_views WHERE viewname LIKE 'undis\_view%' ORDER BY viewname;
 SELECT * FROM another_schema.undis_view3 ORDER BY 1, 2;
 
+-- test if the drop in undistribute can cascade to an extension
+CREATE TABLE extension_table (a INT);
+SELECT create_distributed_table('extension_table', 'a');
+CREATE VIEW extension_view AS SELECT * FROM extension_table;
+ALTER EXTENSION plpgsql ADD VIEW extension_view;
+
+SELECT undistribute_table ('extension_table');
+
```

Comment:
From the codebase it was hard to deduce in which direction the chain was traversed. Would be good to add some extra tests where we have a type that is part of an extension ( I guess Citus does have some type exposed ) would be used in this table. As a negative test, for which the error should not show up.